aws_codeseeder/commands/_seedkit_commands.py

```diff
@@ -96,7 +100,7 @@ def deploy_seedkit(
     if stack_exists:
         deploy_id = stack_outputs.get("DeployId")
         LOGGER.info("Seedkit found with DeployId: %s", deploy_id)
-    template_filename: str = _cfn_seedkit.synth(
+    template_filename: Optional[str] = _cfn_seedkit.synth(
```

Comment:
Would it make more sense to have `synth` return the string template instead of the path?

Right now there's a coupling that `synthesize=False` changes the output value.  That looks like quite a refactor though, so maybe it isn't worth it. Probably just add an assertion under `if not synthesize` that `template_filename` is non-None?

Reply:
Yeah I considered refactoring this but did not want to make too intrusive changes. Good call about an assertion though.